rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,571 @@
+
+### Operator signed in
+At the start of their shift, operators need to confirm that they are fit-for-duty and do not have any electronic devices. They do this by either physically signing a paper trainsheet, or by typing their badge number into Glides. This event represent either type of signin, but may not be generated in the case where an operator only signs into the paper trainsheet.
+
+Event type: `com.mbta.ctd.glides.operator-signed-in`
+Fields in the event:
+- `author` (Author): the inspector who signed in the operator
+- `operator` (Operator): the operator who signed in
+- `signedInAt` (RFC3999 timestamp): the time at which they signed in (separate from the `time` of the event)
+
+### Trip added
+This creates a new trip in the timesheet. It may or may not be a new trip relative to GTFS (see the splitting a train reference example).
+
+Event type: `com.mbta.ctd.glides.trip-added`
+Fields in the event:
+- `author` (Author): the inspector adding the trip
+- `tripKey` (Trip Key): provides a unique identifier for the newly created trip
+- `route` (string, optional): if specified, the route tag for the trip. This is different from a GTFS route ID.
+- `startLocation` (Location, conditionally required): where the trip will be starting. Required if `departureTime` is specified.
+- `endLocation` (Location, conditionally required): where the trip will be ending. Required if `arrivalTime` is specified.
+- `departureTime` (Time, conditionally required): when the added trip is expected to depart `startLocation`.
+- `arrivalTime` (Time, conditionally required): when the added trip is expected to arrive at `endLocation`.
+- `previousTripKey` (Trip Key, optional): links the newly created trip to the trip immediately before it. If the Trip Key refers to an added trip, it may not have been seen in the event stream at the time of this message. Required if `endLocation` is specified and `arrivalTime` is NOT specified.
+- `nextTripKey` (Trip Key, optional): links the newly created trip to the trip after it. If the Trip Key refers to an added trip, it may not have been seen in the event stream at the time of this message. Required if `startLocation` is specified and `departureTime` is NOT specified.
+- `consist` (array of Optional Car, optional, default: empty list): the cars assigned to perform this trip. If a car is `null`, then no car is currently assigned to that position in the consist.
+- `operators` (list of Optional Operator, optional, default: empty list): the array of operators assigned to perform this trip. If an operator is `null`, then no operator is assigned to the car in that position in the consist.
+- `comment` (string, optional): free text with additional information about the trip
+
+*Notes*
+- At least one of `startLocation` and `endLocation` is required.
+- If `startLocation` is 
+- The event should only include values which were explicitly included by the `author`: inferred values should not be included.
```

Comment:
This is good for archiving actions taken in Glides, but it creates the potential for a scenario in which Glides and RTR infer different values and become desynced. I would prefer to avoid the potential for that scenario if we can. Should we add a type for a possibly-inferred value?